sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -83,6 +83,7 @@ object FileFormatWriter extends Logging {
    */
   private[sql] var outputOrderingMatched: Boolean = false
 
+  // scalastyle:off argcount
```

Comment:
nit: we can pass in a wrapper class `PartitionSpec(partitionColumns: Seq[Attribute], numStaticPartitions: Int)` to avoid this.

Reply:
I used a new parameter with default value is for compatible with downstream project as far as possible, does it make sense to you ?